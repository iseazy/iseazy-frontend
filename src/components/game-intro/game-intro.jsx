import { GameIntroIcon } from './game-intro-icon';

export const GameIntro = ({ onClick }) => (
    <div className="game-intro">
        <div className="game-intro__icon">
            <GameIntroIcon />
        </div>
        <h1>MeMemory</h1>
        <button className="game-intro__button" onClick={ onClick }>Comenzar</button>
    </div>
);
