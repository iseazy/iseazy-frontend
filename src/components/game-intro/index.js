export { GameIntro } from './game-intro';
