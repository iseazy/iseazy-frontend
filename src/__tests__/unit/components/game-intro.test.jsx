import { render, fireEvent, screen } from '@testing-library/react';
import { GameIntro } from '../../../components/game-intro/game-intro';

jest.mock('../../../components/game-intro/game-intro-icon', () => ({
    GameIntroIcon: () => 'mockedIcon'
}));

describe('<GameIntro />', () => {
    const testProps = {
        onClick: () => {}
    };
    
    it('should render component', () => {
        const { asFragment } = render(<GameIntro { ...testProps }/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be clicked', () => {
        const onClick = jest.fn();
        render(<GameIntro onClick={ onClick }/>);

        fireEvent.click(screen.getByRole('button'));

        expect(onClick).toHaveBeenCalled();
    });
});
