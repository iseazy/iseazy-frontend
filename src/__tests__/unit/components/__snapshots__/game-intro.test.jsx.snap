// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameIntro /> should render component 1`] = `
<DocumentFragment>
  <div
    class="game-intro"
  >
    <div
      class="game-intro__icon"
    >
      mockedIcon
    </div>
    <h1>
      MeMemory
    </h1>
    <button
      class="game-intro__button"
    >
      Comenzar
    </button>
  </div>
</DocumentFragment>
`;
